src/NDItk/base/InformationRecord.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef NJOY_NDITK_BASE_INFORMATIONRECORD
+#define NJOY_NDITK_BASE_INFORMATIONRECORD
+
+// system includes
+#include <string>
+#include <sstream>
+#include <iomanip>
+
+// other includes
+#include "NDItk/base/SingleValueRecord.hpp"
+
+namespace njoy {
+namespace NDItk {
+namespace base {
+
+/**
+ *  @brief An NDI record with a line of information
+ */
+class InformationRecord : protected SingleValueRecord< InformationRecord, std::string > {
+
+  friend class SingleValueRecord< InformationRecord, std::string >;
+  using Parent = SingleValueRecord< InformationRecord, std::string >;
+
+protected:
+
+  /**
+   *  @brief Write the record data
+   *
+   *  This assumes that the record is not empty.
+   *
+   *  @param[in] iter   the current position in the output
+   */
+  template< typename OutputIterator >
+  void write( OutputIterator& iter ) const {
+
+    unsigned int indent = 4;
+    while ( indent-- ) { *iter++ = ' '; }
+    for ( auto c : this->data().value() ) { *iter++ = c; }
+    *iter++ = '\n';
+  };
+
+  using Parent::key;
+
+public:
+
+  /* constructor */
+
+  /**
+   *  @brief Constructor
+   *
+   *  @param[in] keyword   the keyword of the record
+   */
+  InformationRecord() :
+      SingleValueRecord( Keyword( "info" ) ) {}
+
+  /**
+   *  @brief Constructor
+   *
+   *  @param[in] keyword   the keyword of the record
+   *  @param[in] value     the value of the record
+   */
+  InformationRecord( std::string value ) :
+      SingleValueRecord( Keyword( "info" ), value ) {}
+
+  /* methods */
+
+  using Parent::keyword;
+  using Parent::data;
+  using Parent::empty;
+  using Parent::print;
+
+  /**
+   *  @brief Read the record data
+   *
+   *  @param[in] iter   the current position in the input
+   */
+  template< typename Iterator >
+  void read( Iterator& iter, const Iterator& end ) {
+
+    // move over the first endline
```

Comment:
Would this endline be at the end of the info keyword?

Reply:
In most cases, yes. I do it this way to ensure that there are no additional non-white space characters between the info keyword and the end line (i.e. "info     stuff   \n more information" should fail but "info    \n  more information" should not)